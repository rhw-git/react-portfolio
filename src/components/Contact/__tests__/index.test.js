import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..';

afterEach(cleanup);

describe('Contact component', () => {
  // render base line test
  it('renders', () => {
    render(<ContactForm></ContactForm>);
  });
});
// match snapshot
describe('matches snapshot', () => {
  // arrange
  const { asFragment } = render(<ContactForm></ContactForm>);
  // assert
  expect(asFragment()).toMatchSnapshot();
});
// link visibility
describe('link visibility', () => {
  it('inserts text into contact link', () => {
    //arrange
    const { getByTestId } = render(<ContactForm></ContactForm>);
    //assert
    expect(getByTestId('contact')).toHaveTextContent("LET'S TALK");
  });
});
// button text visibility
describe('button text visibility', () => {
  it('inserts text into button', () => {
    //arrange
    const { getByTestId } = render(<ContactForm></ContactForm>);
    //assert
    expect(getByTestId('button')).toHaveTextContent('Submit');
  });
});
