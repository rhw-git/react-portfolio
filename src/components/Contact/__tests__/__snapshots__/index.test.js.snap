// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="px-4 py-4 container-fluid"
    id="contact-form"
  >
    <form
      class=""
      id="contact-form"
    >
      <div
        class="py-3 text-muted px-0 col-12"
      >
        <p
          data-testid="contact"
        >
          LET'S TALK
        </p>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
          for="name"
        >
          Name
        </label>
        <div
          class="col"
        >
          <input
            class="form-control"
            id="name"
            name="firstName"
            placeholder="First name"
            value=""
          />
        </div>
        <div
          class="col"
        >
          <input
            class="form-control"
            id="name"
            name="lastName"
            placeholder="Last name"
            value=""
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
          for="email"
        >
          Email
        </label>
        <div
          class="col-sm-10"
        >
          <input
            class="form-control"
            id="email"
            name="email"
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        class="pt-3 form-group"
      >
        <label
          class="form-label"
          for="message"
        >
          Message
        </label>
        <textarea
          class="form-control"
          id="message"
          name="message"
          placeholder="Tell me about our next project!"
          rows="5"
        />
      </div>
      <div
        class="form-group row"
      >
        <div
          class="col-sm-10"
        >
          <button
            class="btn btn-outline-secondary"
            data-testid="button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
